bookkeeper-server/src/test/java/org/apache/bookkeeper/bookie/BookieInitializationTest.java

```diff
@@ -781,7 +786,7 @@ public void testPersistBookieStatus() throws Exception {
         bookie = bookieServer.getBookie();
         assertTrue(bookie.isReadOnly());
         // transition to writable mode
-        bookie.doTransitionToWritableMode();
+        ((Bookie.BookieStateManager) bookie.getStateManager()).doTransitionToWritableMode();
```

Comment:
why do we need this kind of cast ?

Reply:
What about making "transitionToWritableMode" return a Future<Void> and than call 
bookie.getStateManager()).transitionToWritableMode().get()
so you won't need to access BookieStateManager ?
//cc @sijie  @ivankelly

Reply:
@eolivelli The `doTransitionToWritableMode()` executed the transition directly and `transitionToWritableMode` will use one `ExecutorService` to run it, I'm trying to keep the original behavior.

Reply:
@ArvinDevel my suggestion will preserve the same behavior without the need of these casts.
Anyway we are in tests some some shortcut is allowed.
I am fine with keeping the casts

Reply:
@ArvinDevel if you make `Bookie#getStateManager()` return `BookieStateManager`, you don't need to cast here.